src/Components/DataCollection/parameter.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState, useEffect } from 'react';
+import { supabase } from '../../supabaseClient';
+import { Link, useParams } from 'react-router-dom';
+import Table from '../Common/CommonComponents/Table';
+import PopUp from '../Common/CommonComponents/PopUp';
+import Button from '../Common/CommonComponents/Button';
+
+export default function Parameter() {
+    const [tableData, setTableData] = useState([]);
+    const { parameter } = useParams();
+    const [isPopupOpen, setIsPopupOpen] = useState(false);
+    const [facilities, setFacilities] = useState([]);
+    const [selectedFacility, setSelectedFacility] = useState(null);
+    const [processes, setProcesses] = useState([]);
+    const [selectedProcess, setSelectedProcess] = useState(null);
+    const [loadingFacilities, setLoadingFacilities] = useState(true);
+    const [loadingProcesses, setLoadingProcesses] = useState(false);
+    const [collectionData, setCollectionData] = useState([]);
+    const [parameterData, setParameterData] = useState([]);
+    const [popupFields, setPopupFields] = useState([]);
+
+    const tableFields = [
+        { id: 'point_name', label: 'Point Name' },
+        { id: 'point_method', label: 'Method' },
+        { id: 'user_name', label: 'User Name' }
+    ];
+
+    useEffect(() => {
+        const fetchFacilities = async () => {
+            setLoadingFacilities(true);
+            const { data, error } = await supabase.from('facility').select('*');
```

Comment:
Add this function to `generalFunction.jsx` and call from there. We are keeping all database functions in that file now.

Reply:
On a second thought we can create an improvement ticket and make these improvements later.